docs/proposals/mux-logging-service.md

```diff
@@ -0,0 +1,305 @@
+[Default Fields](https://github.com/ViaQ/elasticsearch-templates/blob/master/namespaces/_default_.yml)
+in order to maximize the utility of the data, being able to correlate the data
+with data from other deployments and applications, etc.  For application
+specific data, create a namespace with the fields described in one or more
+namespace templates such as found in the
+[namespaces directory](https://github.com/ViaQ/elasticsearch-templates/blob/master/namespaces).
+
+mux Fluentd Config Details
+--------------------------
+
+### Basic Flow ###
+
+With Fluentd configured with `USE_MUX_CLIENT true`, and with mux configured
+With `USE_MUX true` and `USE_MUX_SERVICE false`:
+
+OpenShift Fluentd node collector sends raw records, collected from `json-file`
+or `journald` or both, to the mux service via secure_forward.
+mux processes the logs using the normal OpenShift log filter/enrichment process
+and sends the logs to Elasticsearch.
+
+With `USE_MUX true` and `USE_MUX_SERVICE true`:
+
+In addition to the above, mux will examine the tags of the incoming
+`secure_forward` records, which can come from `secure_forward` clients outside
+of the cluster.  It is expected that the record has the `@timestamp` field and
+is already formatted for the ViaQ data model.  If the tag matches
+`project.namespacename.**`, mux extracts the `namespacename` and passes this to
+the k8s meta filter plugin in order to get the namespace uuid.  If the tag does
+not match, mux uses a namespacename of `mux-undefined`.  Except for the k8s
+meta filter, and the viaq filter, the record is not processed.  The record is
+written to Elasticsearch in the index
+`project.namespacename.namespaceuuid.YYYY.MM.DD`, similar to how OpenShift
+container records are written.
+
+### Details ###
+
+mux *does not* read the logs/journal on the system - it does not mount
+`/var/log`, etc.  This is why it is a DeploymentConfig and not a DaemonSet.  As
+such, mux does not need to mount `/var/log`, `/var/lib`, and `/run/log`.
+However, since it does need to know the absolute time, it still needs to mount
+`/etc/localtime` from the host.
+
+The records come into mux via the secure_forward listener:
+
+    <source>
+      @type secure_forward
+      @label @MUX
+      port "#{ENV['FORWARD_LISTEN_PORT'] || '24284'}"
+      self_hostname "#{ENV['FORWARD_LISTEN_HOST'] || 'mux.example.com'}"
+      shared_key "#{File.open('/etc/fluent/muxkeys/shared_key') do |f| f.readline end.rstrip}"
+      secure yes
+      cert_path /etc/fluent/muxkeys/cert
+      private_key_path /etc/fluent/muxkeys/key
+    </source>
+
+The `logging-mux` secret is mounted on `/etc/fluent/muxkeys` and holds the mux
+TLS server `cert` and `key`, as well as the `ca` cert and the `shared_key`.
+
+These records are sent to the `@MUX` label.  The first thing here is this
+filter:
+
+    <filter kubernetes.var.log.containers.**>
+
+Since mux has no way to know if the Kubernetes records were read from
```

Comment:
Consider making this more an affirmative statement of what we are doing.  Is it significant to know we use one or the other?  I think its more significant to know: what it does, how it varies based on a given condition.